python/ray/autoscaler/_private/fake_multi_node/node_provider.py

```diff
@@ -389,8 +398,20 @@ def _create_node_with_resources_and_labels(
             all_tags.update(tags)
             self._nodes[next_id] = {
```

Comment:
why not set `self._nodes[next_id]` after `self._start_node_delay_s`? Currently, we initialize it and then update it later after `self.start_node_delay_s`.

Reply:
If we only set `self._nodes[next_id]` after `self._start_node_delay_s`, then `non_terminated_nodes` and `is_terminated` will become incorrect.

Reply:
Got it. This makes sense. You are simulating that a node is "allocated" but the ray process is not running.